From ba4386db9f34a645bf85e0b82e28d5f093e9f1d0 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@freescale.com>
Date: Tue, 2 Feb 2016 17:05:01 +0200
Subject: [PATCH 062/146] dpaa_eth: memset the S/G entry structure

Explicitly memset the Scatter-Gather entry structure to make sure there's no
garbage inside its fields.

Signed-off-by: Cristian Sovaiala <cristian.sovaiala@freescale.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
index 1da4f73..3e4c652 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
@@ -742,6 +742,9 @@ int __hot skb_to_sg_fd(struct dpa_priv_s *priv,
 		return -ENOMEM;
 	}
 
+	memset(sgt_buf, 0, priv->tx_headroom +
+		sizeof(struct qm_sg_entry) * (1 + nr_frags));
+
 	/* Enable L3/L4 hardware checksum computation.
 	 *
 	 * We must do this before dma_map_single(DMA_TO_DEVICE), because we may
-- 
2.1.0.27.g96db324

